L1Trigger/L1THGCal/test/HGCalBackendStage1ParameterExtractor.cc

```diff
@@ -0,0 +1,293 @@
+      itr_insert.first->second.emplace(id);
+    }
+  }
+
+  // loop over trigger cells
+  edm::LogPrint("JSONFilling") << "Filling JSON map";
+  for (const auto& triggercell_cells : trigger_cells) {
+    DetId id(triggercell_cells.first);
+    // get module ID and check if relevant module (sector0, zside=-1)
+    uint32_t moduleId = triggerGeometry_->getModuleFromTriggerCell(id);
+    if (moduleId == 0)
+      continue;
+    HGCalTriggerModuleDetId tc_module(moduleId);
+    if (!(tc_module.isHGCalModuleDetId()) || (tc_module.zside() < 0) || (tc_module.sector() != 0))
+      continue;
+
+    // check that module is connected to a stage 1 FPGA
+    HGCalTriggerGeometryBase::geom_set lpgbts = triggerGeometry_->getLpgbtsFromModule(moduleId);
+    if (lpgbts.size() == 0)
+      continue;
+
+    // only retrieve mapping for the tested fpga
+    uint32_t fpgaId = triggerGeometry_->getStage1FpgaFromModule(moduleId);
+    HGCalTriggerBackendDetId tc_fpga(fpgaId);
+    if (!(tc_fpga.isStage1FPGA()) || (tc_fpga.sector() != 0) || (tc_fpga.zside() < 0))
+      continue;
+    int fpgaLabel = tc_fpga.label();
+    if (fpgaLabel != the_fpga_)
+      continue;
+
+    // retrieve information to be saved
+    int triggerCellSubdet = 0;
+    int triggerCellLayer = 0;
+    int triggerCellUEta = 0;
+    int triggerCellVPhi = 0;
+    if (id.det() == DetId::HGCalHSc) {
+      HGCScintillatorDetId id_sc(triggercell_cells.first);
+      triggerCellSubdet = id_sc.subdet();
+      triggerCellLayer = id_sc.layer();
+      triggerCellUEta = id_sc.ietaAbs();
+      triggerCellVPhi = id_sc.iphi();
+    } else if (HFNoseTriggerDetId(triggercell_cells.first).det() == DetId::HGCalTrigger &&
+               HFNoseTriggerDetId(triggercell_cells.first).subdet() == HGCalTriggerSubdetector::HFNoseTrigger) {
+      HFNoseTriggerDetId id_nose_trig(triggercell_cells.first);
+      triggerCellSubdet = id_nose_trig.subdet();
+      triggerCellLayer = id_nose_trig.layer();
+      triggerCellUEta = id_nose_trig.triggerCellU();
+      triggerCellVPhi = id_nose_trig.triggerCellV();
+    } else {
+      HGCalTriggerDetId id_si_trig(triggercell_cells.first);
+      triggerCellSubdet = id_si_trig.subdet();
+      triggerCellLayer = id_si_trig.layer();
+      triggerCellUEta = id_si_trig.triggerCellU();
+      triggerCellVPhi = id_si_trig.triggerCellV();
+    }
+    GlobalPoint position = triggerGeometry_->getTriggerCellPosition(triggercell_cells.first);
+    float triggerCellX = position.x();
+    float triggerCellY = position.y();
+    float triggerCellZ = position.z();
+    float triggerCellEta = position.eta();
+    float triggerCellPhi = position.phi();
+    float triggerCellRoverZ = sqrt(triggerCellX * triggerCellX + triggerCellY * triggerCellY) / triggerCellZ;
+
+    // transform HGCalHSc TC coordinates to define a TC address in [0,47]
+    uint32_t tce = triggerCellUEta;
+    uint32_t tcp = triggerCellVPhi;
+    if (triggerCellSubdet == 10) {  // HGCalHSc
```

Comment:
Would be better to use the trigger tools:
https://github.com/hgc-tpg/cmssw/blob/9bff0b25ff7a1ebf947c8ef0043ea00e35efb07c/L1Trigger/L1THGCal/interface/HGCalTriggerTools.h#L45

Reply:
Done